pilot/pkg/proxy/envoy/v2/cds.go

```diff
@@ -45,26 +46,34 @@ func (conn *XdsConnection) clusters(response []*xdsapi.Cluster) *xdsapi.Discover
 	return out
 }
 
-func (s *DiscoveryServer) pushCds(con *XdsConnection, push *model.PushContext, version string) error {
+func (s *DiscoveryServer) pushCds(con *XdsConnection, push *model.PushContext, version string) (err error) {
 	// TODO: Modify interface to take services, and config instead of making library query registry
+	pushStart := time.Now()
+	defer func() {
+		if err != nil {
+			cdsPushErrTime.Record(time.Since(pushStart).Seconds())
```

Comment:
Isn't seconds too much? milli-second

Reply:
will do

Reply:
can you do whatever the proxy convergence one does? not sure what that is

Reply:
@hzxuzhonghu @howardjohn proxy convergence uses `Seconds`. I'm looking at our metrics and think `Seconds` should be reasonable for this case:
```
# HELP pilot_xds_push_time Total time Pilot takes to push lds, rds, cds and eds.
# TYPE pilot_xds_push_time histogram
pilot_xds_push_time_bucket{type="cds",le="1"} 204
pilot_xds_push_time_bucket{type="cds",le="3"} 204
pilot_xds_push_time_bucket{type="cds",le="5"} 204
pilot_xds_push_time_bucket{type="cds",le="10"} 204
pilot_xds_push_time_bucket{type="cds",le="20"} 204
pilot_xds_push_time_bucket{type="cds",le="30"} 204
pilot_xds_push_time_bucket{type="cds",le="50"} 204
pilot_xds_push_time_bucket{type="cds",le="100"} 204
pilot_xds_push_time_bucket{type="cds",le="+Inf"} 204
pilot_xds_push_time_sum{type="cds"} 29.64185306899997
pilot_xds_push_time_count{type="cds"} 204
pilot_xds_push_time_bucket{type="eds",le="1"} 1497
pilot_xds_push_time_bucket{type="eds",le="3"} 1701
pilot_xds_push_time_bucket{type="eds",le="5"} 1705
pilot_xds_push_time_bucket{type="eds",le="10"} 1705
pilot_xds_push_time_bucket{type="eds",le="20"} 1705
pilot_xds_push_time_bucket{type="eds",le="30"} 1705
pilot_xds_push_time_bucket{type="eds",le="50"} 1705
pilot_xds_push_time_bucket{type="eds",le="100"} 1705
pilot_xds_push_time_bucket{type="eds",le="+Inf"} 1705
pilot_xds_push_time_sum{type="eds"} 354.7408447529996
pilot_xds_push_time_count{type="eds"} 1705
pilot_xds_push_time_bucket{type="lds",le="1"} 203
pilot_xds_push_time_bucket{type="lds",le="3"} 203
pilot_xds_push_time_bucket{type="lds",le="5"} 204
pilot_xds_push_time_bucket{type="lds",le="10"} 204
pilot_xds_push_time_bucket{type="lds",le="20"} 204
pilot_xds_push_time_bucket{type="lds",le="30"} 204
pilot_xds_push_time_bucket{type="lds",le="50"} 204
pilot_xds_push_time_bucket{type="lds",le="100"} 204
pilot_xds_push_time_bucket{type="lds",le="+Inf"} 204
pilot_xds_push_time_sum{type="lds"} 14.685711940000003
pilot_xds_push_time_count{type="lds"} 204
pilot_xds_push_time_bucket{type="rds",le="1"} 0
pilot_xds_push_time_bucket{type="rds",le="3"} 0
pilot_xds_push_time_bucket{type="rds",le="5"} 0
pilot_xds_push_time_bucket{type="rds",le="10"} 0
pilot_xds_push_time_bucket{type="rds",le="20"} 0
pilot_xds_push_time_bucket{type="rds",le="30"} 0
pilot_xds_push_time_bucket{type="rds",le="50"} 204
pilot_xds_push_time_bucket{type="rds",le="100"} 204
pilot_xds_push_time_bucket{type="rds",le="+Inf"} 204
pilot_xds_push_time_sum{type="rds"} 7863.3871518669985
pilot_xds_push_time_count{type="rds"} 204
```

Maybe we can redefine the bucket number/ceiling? The current ones I'm using are the same as of proxy convergence. What do you guys think?